internal/core/src/query/SearchBruteForce.cpp

```diff
@@ -54,6 +56,38 @@ BruteForceSearch(const dataset::SearchDataset& dataset,
         auto base_dataset =
             knowhere::GenDataSet(chunk_rows, dim, chunk_data_raw);
         auto query_dataset = knowhere::GenDataSet(nq, dim, dataset.query_data);
+
+        if (data_type == DataType::VECTOR_FLOAT16) {
+            // Todo: Temporarily use cast to float32 to achieve, need to optimize
+            // first, First, transfer the cast to knowhere part
+            // second, knowhere partially supports float16 and removes the forced conversion to float32
+            auto xb = base_dataset->GetTensor();
+            std::vector<float> float_xb(base_dataset->GetRows() *
```

Comment:
1. create it only if it's fp16 vector.
2. use smart pointer do you don't need to manually maintain life cycle

Reply:
Thanks for your correction, I'm still thinking about an appropriate way to modify the BruteForceSearch function.

Reply:
> Thanks for your correction, I'm still thinking about an appropriate way to modify the BruteForceSearch function.

but this might reduce actual work even if it's not Float16 vectors right?

Reply:
Probably not. If it's not float16 vectors, the original program will be retained.

Reply:
> Probably not. If it's not float16 vectors, the original program will be retained.

that's what I'm talking about.
for non float16 vectors, float_xb is allocated for no reason

Reply:
float_xb is surrounded by [if (data_type == DataType::VECTOR_FLOAT16) {}]. If it is not float16, this part will not be executed. I'm not sure the memory is allocated at runtime.  Will this cause memory problems?